torchao/core/config.py

```diff
@@ -284,3 +296,28 @@ def config_from_dict(data: Dict[str, Any]) -> AOBaseConfig:
         return cls(**processed_data)
     except Exception as e:
         raise ValueError(f"Failed to create instance of {cls.__name__}: {e}")
+
+@dataclass
+class Float8TypeConfig:
+    """
+    Configuration for selecting the preferred float8 type pair, either e4m3fn/e5m2 or e4m3fnuz/e5m2fnuz.
+
+    Currently, ROCm supports 1. fnuz variants in MI300. 2. OCP F8 variants in MI350/Navi4.
+    """
+
+    # The preferred e4m3 type.
+    e4m3_dtype = torch.float8_e4m3fn
+
+    # The preferred e5m2 type.
+    e5m2_dtype = torch.float8_e5m2
+
+    def __post_init__(self):
+        if torch.version.hip and torch.cuda.is_available() and is_MI300():
```

Comment:
I don't like this, I think magic methods should be avoided. It isn't super clear what would happen here, for example, if someone saves a checkpoint on an nvidia device and loads it on a rocm device which needs these other flavors.

I would recommend making the user change the float8 dtypes to the fnuz variants explicitly.

Reply:
>  It isn't super clear what would happen here, for example, if someone saves a checkpoint on an nvidia device and loads it on a rocm device which needs these other flavors.

Just FYI this errors currently for torchao fp8, as the weight tensors will be loaded in the ocp variant (from the checkpoint), while the activation tensors will be the fnuz variant. 

> would it complicate the user API? then they'd need to be aware whether they are quantizing for CUDA or ROCm?

I think yes, but this is really a MI300x only problem - they only have support for fnuz fp8, MI350x will have support for OCP fp8. So maybe the solution here is to just throw a warning / not implemented error when we detecting using ocp fp8 on MI300x.

Reply:
OK, I have reverted this change and have been using the dtype from `torch` namespace directly